### PR TITLE
Minor platform benchmarks improvements

### PR DESCRIPTION
**Pull request by adamsitnik**
It gives around +20k RPS for both JSON and PlainText

@sebastienros PTAL

### CODE REVIEW

File: src/BenchmarksApps/Kestrel/PlatformBenchmarks/BenchmarkApplication.cs [hunk 1]
```diff
@@ -116,20 +118,23 @@ private static void Json(PipeWriter pipeWriter)
...
+            writer.WriteNumeric((uint)JsonPayload.Length);
 
             // End of headers
             writer.Write(_eoh);
+            writer.Commit();
 
             // Body
-            writer.Write(jsonPayload);
-            writer.Commit();
+            using (var utf8jsonWriter = new Utf8JsonWriter(writer.Output))
```
Comment (adamsitnik) [inline src/BenchmarksApps/Kestrel/PlatformBenchmarks/BenchmarkApplication.cs]: the main difference for using `Utf8JsonWriter` is that we don't allocate an array of bytes

File: src/BenchmarksApps/Kestrel/PlatformBenchmarks/BenchmarkApplication.cs [hunk 2]
```diff
@@ -27,6 +27,8 @@ public partial class BenchmarkApplication
         private readonly static AsciiString _plainTextBody = "Hello, World!";
 
         private static readonly JsonSerializerOptions SerializerOptions = new JsonSerializerOptions();
+        private static readonly JsonMessage _jsonMessage = new JsonMessage { message = "Hello, World!" };
```
Comment (benaadams) [inline src/BenchmarksApps/Kestrel/PlatformBenchmarks/BenchmarkApplication.cs]: Doesn't meet the requirements https://github.com/TechEmpower/FrameworkBenchmarks/wiki/Project-Information-Framework-Tests-Overview#requirements
```
i. For each request, an object mapping the key message to Hello, World! must be instantiated.
```

File: src/BenchmarksApps/Kestrel/PlatformBenchmarks/BufferExtensions.cs [hunk 3]
```diff
@@ -26,34 +26,32 @@ internal static unsafe void WriteNumeric<T>(ref this BufferWriter<T> buffer, uin
             const byte AsciiDigitStart = (byte)'0';
 
             var span = buffer.Span;
-            var bytesLeftInBlock = span.Length;
 
             // Fast path, try copying to the available memory directly
             var advanceBy = 0;
-            fixed (byte* output = span)
+            if (span.Length >= 3)
```
Comment (adamsitnik) [inline src/BenchmarksApps/Kestrel/PlatformBenchmarks/BufferExtensions.cs]: there is no need to pin this span, with size hint we always execute the "happy path" with size big enough and no boundaries check when indexing

File: src/BenchmarksApps/Kestrel/PlatformBenchmarks/BenchmarkApplication.cs (hunk 2 above)
Comment (adamsitnik) [reply src/BenchmarksApps/Kestrel/PlatformBenchmarks/BenchmarkApplication.cs]: thanks for the catch, I am going to fix it right away

**Comment by benaadams** [comment]
Has changed a little, but I think this is now included in https://github.com/aspnet/Benchmarks/pull/1473 and https://github.com/TechEmpower/FrameworkBenchmarks/pull/5579; if you want to double check?

**Comment by adamsitnik** [comment]
I am going to send a new PR

**Pull request closed without merging**